src/engine/TPA.cc

```diff
@@ -1769,12 +1782,10 @@ PTRef TPABase::getSafetyExplanation() const {
         // TODO: compute the safe inductive invariant and return negation of that?
         return init;
     }
-    auto power = explanation.power;
-    PTRef transitionInvariant = explanation.relationType == TPAType::LESS_THAN
-                                    ? getPower(power, explanation.relationType)
-                                    : logic.mkOr(shiftOnlyNextVars(getPower(power, TPAType::LESS_THAN)),
-                                                 logic.mkAnd(getPower(power, TPAType::LESS_THAN),
-                                                             getNextVersion(getPower(power, TPAType::EQUALS))));
```

Comment:
Here was the problem before. This computation was not correct if we got here after with concluded that the system is safe because of (left or right) invariants computed from the Houdini check.

Reply:
Hmm, Ok, you basically reworked the data structure as I can see...

Reply:
Yes, instead of remembering which abstract relation is the transition invariant, we store the transition invariant directly.
Instead of `power`, which indicated which abstract relation should be used, we now remember the value `k` for which the transition invariant is k-inductive. Or rather, we store the exponent of the power of 2. That is, we store `exp` so that the invariant is `2^exp`-inductive.